internal/service/connect/hours_of_operation.go

```diff
@@ -0,0 +1,352 @@
+									},
+									"minutes": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+						"start_time": {
+							Type:     schema.TypeList,
+							MaxItems: 1,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"hours": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+									"minutes": {
+										Type:     schema.TypeInt,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+				Set: func(v interface{}) int {
+					var buf bytes.Buffer
+					m := v.(map[string]interface{})
+					buf.WriteString(m["day"].(string))
+					buf.WriteString(fmt.Sprintf("%+v", m["end_time"].([]interface{})))
+					buf.WriteString(fmt.Sprintf("%+v", m["start_time"].([]interface{})))
+					return create.StringHashcode(buf.String())
+				},
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 250),
+			},
+			"hours_of_operation_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hours_of_operation_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 127),
+			},
+			"tags":     tftags.TagsSchema(),
+			"tags_all": tftags.TagsSchemaComputed(),
+			"time_zone": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceHoursOfOperationCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conn := meta.(*conns.AWSClient).ConnectConn
+	defaultTagsConfig := meta.(*conns.AWSClient).DefaultTagsConfig
+	tags := defaultTagsConfig.MergeTags(tftags.New(d.Get("tags").(map[string]interface{})))
+
+	instanceID := d.Get("instance_id").(string)
+	name := d.Get("name").(string)
+
+	config := expandConfigs(d.Get("config").(*schema.Set).List())
+
+	input := &connect.CreateHoursOfOperationInput{
+		Config:     config,
+		InstanceId: aws.String(instanceID),
+		Name:       aws.String(name),
+		TimeZone:   aws.String(d.Get("time_zone").(string)),
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		input.Description = aws.String(v.(string))
+	}
+
+	if len(tags) > 0 {
+		input.Tags = Tags(tags.IgnoreAWS())
+	}
+
+	log.Printf("[DEBUG] Creating Connect Hours of Operation %s", input)
+	output, err := conn.CreateHoursOfOperationWithContext(ctx, input)
+
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error creating Connect Hours of Operation (%s): %w", name, err))
+	}
+
+	if output == nil {
+		return diag.FromErr(fmt.Errorf("error creating Connect Hours of Operation (%s): empty output", name))
+	}
+
+	d.SetId(fmt.Sprintf("%s:%s", instanceID, aws.StringValue(output.HoursOfOperationId)))
```

Comment:
Standard in the code base is to use `decode` and `encode` please move this logic to a function `encode`. here is an example: https://github.com/hashicorp/terraform-provider-aws/blob/db99355e2cf1edbb8fb14ec87ee5f66dbfa58236/internal/service/ec2/vpc_ipam_pool_cidr.go#L98

Reply:
After discussions, for consistency with other connect resources, we will retain the current `fmt.Sprintf` approach.

https://github.com/hashicorp/terraform-provider-aws/blob/b5f182feced587e2715478c045ef5f78eba7f84d/internal/service/connect/contact_flow.go#L141